### PR TITLE
Add jwt access credentials

### PR DESCRIPTION
**Pull request by stanley-cheung**
* Add ServiceAccountJwtAccessCredentials class
 * Change updateMetadata callback function signature to accept authUri parameter
 * if no scope is set in ServiceAccountCredentials, use ServiceAccountJwtAccessCredentials instead
 * to calculate the token for the ServiceAccountJwtAccessCredentials class, use the passed in authUri to encode the JWT. no outbound call needs to be made
 * add some unit tests

### CODE REVIEW

**Comment by stanley-cheung** [comment]
Fixes issue #43

**Comment by stanley-cheung** [comment]
Please review, thanks!

File: src/ServiceAccountCredentials.php [hunk 1]
```diff
@@ -104,4 +104,102 @@ public function getCacheKey()
...
+/**
+ * Authenticates requests using Google's Service Account credentials via
+ * JWT Access.
+ *
+ * This class allows authorizing requests for service accounts directly
+ * from credentials from a json key file downloaded from the developer
+ * console (via 'Generate new Json Key').  It is not part of any OAuth2
+ * flow, rather it creates a JWT and sends that as a credential.
+ */
+class ServiceAccountJwtAccessCredentials extends CredentialsLoader
```
Comment (kesar) [inline src/ServiceAccountCredentials.php]: I would add this class in a different file. At the same time now that there are two different ServiceAccountCredentials an interface could make sense with the method  updateMetadata

Comment (tbetbetbe) [inline src/ServiceAccountCredentials.php]: +1 for having a separate class.

**Comment by anthmgoogle** [comment]
Yes, this looks already done in this case.

**Pull request merged**